### PR TITLE
Fix and re-enable cppcheck

### LINKED ISSUE #188: Re-enable cppcheck
In commit a91f8f9 (PR #187 ) I had to disable cppcheck, because it failed for no good reason.

I have no idea of what's happening, we should try again in some days/weeks and maybe re-enable it.

### PR DESCRIPTION
**Pull request by fangerer**
Resolves #188 .

This fixes and re-enables the `cppcheck` job.
I think the main reason why it broke before is that there was an update of the Ubuntu runner image (`ubuntu-latest` was updated to `ubuntu-22.04`) which also includes a newer cppcheck version.

Funnily, I first tried to reproduce the problem on Ubuntu 20.04 with cppcheck 1.90 and did fixes to make that run. Without changing HPy's source code, this did not pass on cppcheck 2.7 (which is available in Ubuntu 22.04). 
Long story short: the cppcheck job broke because of cppcheck. 

I therefore pinned the versions to Ubuntu 22.04 and cppcheck 2.7-1. This should work for a while (a few years).

### CODE REVIEW

**Comment by mattip** [comment]
CI is green, let's give this a try. Thanks @fangerer
